src/lib.rs

```diff
@@ -179,31 +210,66 @@ impl GpuProfiler {
 
             size_for_new_query_pools: QueryPool::MIN_CAPACITY,
 
-            max_num_pending_frames,
-            timestamp_to_sec: timestamp_period as f64 / 1000.0 / 1000.0 / 1000.0,
+            settings,
+
+            device_features: None,
 
             #[cfg(feature = "tracy")]
-            tracy_context: tracy::create_tracy_gpu_client(_adapter.get_info().backend, device, queue, timestamp_period),
-        }
+            tracy_context: None,
+        })
     }
 
-    /// Returns true if a timestamp should be written to the encoder or pass.
-    fn timestamp_write_enabled<Recorder: ProfilerCommandRecorder>(&self, encoder_or_pass: &mut Recorder) -> bool {
-        (encoder_or_pass.is_pass() && self.enable_pass_timer) || (!encoder_or_pass.is_pass() && self.enable_encoder_timer)
+    /// Creates a new profiler and connects to a running Tracy client.
+    #[cfg(feature = "tracy")]
+    pub fn new_with_tracy_client(
```

Comment:
The separate method for running with tracy certainly allows for the the most flexibility for handling any Tracy errors ,and for controlling whether it is enabled at runtime, although it does make things more awkward for the just always running with tracy if the corresponding feature is enabled.

Mainly just an observation, I don't think anything necessarily needs to be changed.